lib/matplotlib/axes/_axes.py

```diff
@@ -2604,6 +2605,123 @@ def barh(self, y, width, height=0.8, left=None, *, align="center",
                            align=align, **kwargs)
         return patches
 
+    def bar_label(self, container, captions=[], *, fmt="%g", position="edge",
+                  padding=0, shifting=0, **kwargs):
+        """
+        Label a bar plot.
+
+        Adds labels to bars in the given `.BarContainer`.
+        You may need to adjust the axis limits to fit the labels.
+
+        Parameters
+        ----------
+        container : `.BarContainer`
+            Container with all the bars and optionally errorbars.
+
+        captions : array-like, optional
```

Comment:
Sorry I am coming in a bit late with naming issues, but would this be better named "labels"?  Else where we use the `label` kwarg / `set_label` method on Artists to control what goes into the legend and the method is called `bar_labels`.